### ISSUE TITLE
Issue: false positive for couldNotResolve for type only package

### ISSUE THREAD

**Issue by mx-bernhard** [sverweij/dependency-cruiser#900]
When using `not-to-unresolvable` from the generated cruiser-config but with `tsPreCompilationDeps: true`, the rule does not see, that the referred type-only package is indeed in the _package.json_ — this is just a theory, but I made a reproduction. This can be observed for example for the package [@chevrotain/types](https://www.npmjs.com/package/@chevrotain/types).

The rule in _.dependency-cruiser.js_:

```js
    {
      name: 'not-to-unresolvable',
      comment:
        "This module depends on a module that cannot be found ('resolved to disk'). If it's an npm " +
        'module: add it to your package.json. In all other cases you likely already know what to do.',
      severity: 'error',
      from: {},
      to: {
        couldNotResolve: true
      }
    }
```

### Expected Behavior

`not-to-unresolvable` should not report any violations when type imports from the package `@chevrotain/types` are consumed when the _package.json_ does indeed contain a `devDependencies`-entry for said package.

### Current Behavior

The following error is emitted:

```log
  error not-to-unresolvable: src/foo.ts → @chevrotain/types

✘ 1 dependency violations (1 errors, 0 warnings). 2 modules, 1 dependencies cruised.
```

### Steps to Reproduce (for bugs)

See [Git-Repo](https://github.com/mx-bernhard/dep-cruise-not-to-unresolvable)

Steps are in [README](https://github.com/mx-bernhard/dep-cruise-not-to-unresolvable/blob/main/README.md)

### Context

We want to enforce specific Typescript-type usage since, e.g., there exist multiple packages for GraphQL-types which seem to contain similar but subtle different typings due to version differences.  But when enabling `tsPreCompilationDeps` in dependency-cruiser, lots of false positives are reported (not just the one above).

### Your Environment

- Version used: 16.0.0
- Node version: 18.18.2
- Operating System and version: WSL 2 with Ubuntu 22.04
- Link to project: https://github.com/mx-bernhard/dep-cruise-not-to-unresolvable (just a repro)

**Comment by mx-bernhard** [comment]
I had looked at conditions in general, but somehow I didn't consider conditionNames. 😀
Thx!

**Comment by sverweij** [comment]
> I had looked at conditions in general, but somehow I didn't consider conditionNames. 😀 Thx!

Me neither t.b.h. - it's been a while I encountered such a package.

After digging around I found the issue - the type only package `@chevrotrain` is of `"type": "module"` _and_ specifies types as an export condition in its package.json (apart from the `type` field). 

```json
  // ...
  "exports": {
    ".": {
      "types": "./api.d.ts"
    }
  },
  // ...
```

I'll make sure the _--init_ template will contain that condition by default, so when a next person is setting up dependency-cruiser they won't encounter this problem.


> b.t.w. @mx-bernhard apart from taking the time to raise this issue - thanks for the excellent reproduction repository!

**Comment by mx-bernhard** [comment]
Is there an option to enable debug/trace-logging of enhanced-resolve so that I can track down issues myself? Currently, I have another issue when adding the entry to conditionNames but with another package [@devexpress/dx-grid-core](https://www.npmjs.com/package/@devexpress/dx-grid-core) which now fails to resolve. See [commit](https://github.com/mx-bernhard/dep-cruise-not-to-unresolvable/commit/46493a7d90238af11e9ee03aae2b0c14b803ac1a). Output:

```log

  error not-to-unresolvable: src/foo.ts → @devexpress/dx-grid-core

✘ 1 dependency violations (1 errors, 0 warnings). 3 modules, 2 dependencies cruised.
```

**Comment by mx-bernhard** [comment]
I took a peek at `enhanced-resolve`, and it seems that [this](https://github.com/webpack/enhanced-resolve/blob/main/lib/Resolver.js#L375) could be a possibility to enable logging. I am guessing, that `dependency-cruiser` is supplying a `resolveContext` to `enhanced-resolve`, right?

Also, I created an issue with my support account on the devexpress site: https://supportcenter.devexpress.com/ticket/details/t1211399/devexpress-dx-grid-core-has-package-json-with-type-definition-exports-that-are-not Maybe they can resolve this quickly.

In the meantime, I have added an exception for the false positive dependency-cruiser reports and consider this issue resolved since it is not a bug of dependency-cruiser. Thank you very much for your assistance on this!
